pytorch_lightning/trainer/trainer.py

```diff
@@ -721,6 +727,7 @@ def fit(
 
             datamodule: An instance of :class:`~pytorch_lightning.core.datamodule.LightningDataModule`.
         """
+        self.training_type_plugin.model = model
```

Comment:
n00b question: why model vs lightning_module?

Reply:
I believe this was to differentiate with `self.lightning_module`, but it might be confusing to the readers.

Reply:
`TTP.lightning_module` doesn't have a setter, `TTP.model` has a setter. 
Initially, `TTP.lightning_module == TTP.model`, and later during setup TTP.model will hold the wrapped model.
This line that I added here (and in the other Trainer methods) is needed for the case when:

```py
trainer.fit(model)
trainer.test()  # no model provided, need to reuse self.lightning_module

```
For spawn plugins, the model reference would be lost if we didn't save it in the main process.